routes/dashboard.rb

```diff
@@ -1,15 +1,34 @@
 class App < Sinatra::Base 
   get '/' do
     @cities = City.all
-    
-    @routes = Route.all.map do |route|
+
+    slim :index
+  end
+
+  post '/' do
+    if params[:departure_id].nil? || params[:departure_id].empty? || params[:arrival_id].nil? || params[:arrival_id].empty?
```

Comment:
Extract to a Policy Object
http://www.eq8.eu/blogs/41-policy-objects-in-ruby-on-rails